lexos/models/topword_model.py

```diff
@@ -0,0 +1,377 @@
+                    data=group_data.loc[group_two_label],
+                    index=words,
+                    name='Class "%s"' % group_two_label))
+            for group_one_label, group_two_label in label_combinations]
+
+        return readable_result
+
+    def _get_result(self, class_division_map: pd.DataFrame) -> TopwordResult:
+        """Call the right method corresponding to user's selection.
+
+        :return: a namedtuple that holds the topword result, which contains a
+                 header and a list of pandas series."""
+
+        if self._topword_front_end_option.analysis_option == "allToPara":
+            # Get header and result.
+            header = "Compare Each Document to All the Documents As a Whole"
+            results = self._analyze_file_to_all()
+
+            return TopwordResult(header=header, results=results)
+
+        elif self._topword_front_end_option.analysis_option == "classToPara":
+            # Check if more than one class exists.
+            assert class_division_map.shape[0] > 1, NOT_ENOUGH_CLASSES_MESSAGE
+
+            # Get header and result.
+            header = "Compare Each Document to Other Class(es)"
+            results = self._analyze_class_to_all(
+                class_division_map=class_division_map)
+
+            return TopwordResult(header=header, results=results)
+
+        elif self._topword_front_end_option.analysis_option == "classToClass":
+            # Check if more than one class exists.
+            assert class_division_map.shape[0] > 1, NOT_ENOUGH_CLASSES_MESSAGE
+
+            # Get header and result.
+            header = "Compare a Class to Each Other Class"
+            results = self._analyze_class_to_class(
+                class_division_map=class_division_map)
+
+            return TopwordResult(header=header, results=results)
+
+        else:
+            raise ValueError("Invalid topword analysis option.")
+
+    def get_readable_result(self, class_division_map: pd.DataFrame) -> \
+            TopwordResult:
+        """Gets the readable result to display on the web page.
+
+        :return: a namedtuple that holds the topword result, which contains a
+                 header and a list of pandas series, however it will check the
+                 length of each pandas series and only return the first 20 rows
+                 if the pandas series has length that is longer than 20."""
+
+        topword_result = \
+            self._get_result(class_division_map=class_division_map)
+        readable_result = [result if result.size <= 20 else result[:20]
+                           for result in topword_result.results]
+
+        return TopwordResult(header=topword_result.header,
+                             results=readable_result)
+
+    def get_topword_csv_path(self, class_division_map: pd.DataFrame) -> str:
+        """Writes the generated top word results to an output CSV file.
+
+        :returns: path of the generated CSV file.
```

Comment:
Seriously, I think Pycharm would complain if you have missing parameter documentation...